### PR TITLE
[10.0.x] Backport NOTICE and LISENSE refactoring and rat-excludes to 10.0.x

### PR DESCRIPTION
**Pull request by tkobayas**
**Ports** 
Backport:
* #6149
* #6153 
* #6164 
* #6156 

**Issue**: 

* https://github.com/apache/incubator-kie-issues/issues/1616

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -200,3 +200,154 @@
...
+only if the software compliance is of a nature fully matching the applicable compliance points as stated in the
+specification. Software developed only partially matching the applicable compliance points may claim only that the
+software was based on this specification, but may not claim compliance or conformance with this specification. In the
+event that testing suites are implemented or approved by Object Management Group, Inc., software developed using this
+specification may claim compliance or conformance with the specification only if the software satisfactorily completes
+the testing suites.
+
+ ------------------------------------------------------------------------------------
+ for drools-model/drools-mvel-parser/src/main/javacc/mvel.jj
+     JavaParser is dual licensed under LGPL license and Apache License Version 2.0 license. We choose the Apache License Version 2.0.
```
Comment (tkobayas) [inline LICENSE:353]: LICENSE is a little different from main --- JavaParser.g4 and JavaLexer.g4 don't exist in 10.0.x.

**Comment by tkobayas** [comment]
@porcelli  you wrote:

https://github.com/apache/incubator-kie-kogito-images/pull/1811#issuecomment-2489695430

> and for 10.0.0. release the individual repos content like .rat-exclude, DISCLAIMER, LEGAL, etc... will be removed for source anyway.

Backport of NOTICE etc. may not be required, but still license headers fixes are required (not sure if it's a blocker). Is it fine to merge this PR?

If this PR is not actually needed to get 10.0.0 released, feel free to close.

File: NOTICE [hunk 2]
```diff
@@ -5,19 +5,9 @@ This product includes software developed at
...
-  Downloaded from: https://lunrjs.com/
-  License: MIT
-
-* search-ui
-  Downloaded from: https://gitlab.com/antora/antora-lunr-extension
-  License: Mozilla Public License 2.0
-
-* antora ui-bundle
-  Downloaded from: https://github.com/stephengold/antora-ui-bundle
-  License: Mozilla Public License 2.0
```
Comment (jomarko) [inline NOTICE:23]: I think we need to add corresponding entries into LICENSE file, as the files are still delivered as part of `apache-kie-10.0.0-incubating-sources.zip`.

File: .rat-excludes [hunk 3]
```diff
@@ -0,0 +1,46 @@
+\.rat-excludes
+\.gitignore
+DISCLAIMER-WIP
+LICENSE-.*\.txt
```
Comment (jomarko) [inline .rat-excludes]: I think we do not have such file `LICENSE-.*\.txt`

File: .rat-excludes [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+\.git-blame-ignore-revs
+dependency-reduced-pom\.xml
+lunr-2\.3\.9\.min\.js
+search-ui\.js
+search\.css
+branch\.yaml
+main\.yaml
+mvel\.jj
+JavaLexer\.g4
+JavaParser\.g4
```
Comment (jomarko) [inline .rat-excludes]: I think we do not have files with `g4` extension

File: .rat-excludes [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+.*\.csv
+.*\.sdo
+.*\.sdt
+TestingEmptyFile\.txt
+.*\.test_json
+.*\.pmml_json
+.*\.drl_json
+.*\.dslr
+.*DMN.*\.xsd
+dmn.xsd
```
Comment (jomarko) [inline .rat-excludes]: I thought `dmn.xsd` is developed as part of KIE community, shouldn't we keep license header for it? I do not see a link for such file on the OMG website.

image

File: .rat-excludes [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+checkstyle-cachefile
+test.*\.txt
+test.*\.yml
+drl\.ftl
+goal\.txt
+adult\.txt
+antora\.yml
+.*\.iml
+CsvLineParser\.java
+target/*
```
Comment (jomarko) [inline .rat-excludes]: Reading the comment [here](https://github.com/apache/incubator-kie-tools/pull/2746#discussion_r1849492240), I thought `target/*` is not valid, but probably it is? Running rat tool locally didn't complain.

**Review by jomarko — COMMENTED** [review]
Thank you for the PR. Some comments inline.

File: .rat-excludes (hunk 6 above)
Comment (jomarko) [reply .rat-excludes]: answer here https://github.com/apache/incubator-kie-tools/pull/2746#discussion_r1851529707

File: NOTICE (hunk 2 above)
Comment (tkobayas) [reply NOTICE:23]: @jomarko Thank you for reviewing! These 3 components are placed under `drools-docs`, which will be excluded by the release CI starting with the next RC build, so we will not need to address them in LICENSE.

File: .rat-excludes (hunk 3 above)
Comment (tkobayas) [reply .rat-excludes]: Thanks, I'll fix.

File: .rat-excludes (hunk 4 above)
Comment (tkobayas) [reply .rat-excludes]: Thanks, I'll fix.

File: .rat-excludes (hunk 5 above)
Comment (tkobayas) [reply .rat-excludes]: The `dmn.xsd` is the same as https://www.omg.org/spec/DMN/20151101/dmn.xsd

File: .rat-excludes [hunk 7]
```diff
@@ -0,0 +1,338 @@
+# .git-blame-ignore-revs
+.git-blame-ignore-revs
```
Comment (tkobayas) [inline .rat-excludes:2]: Applied "list every single file" style to align with other repos. It's easier to review.

For main, filed https://github.com/apache/incubator-kie-drools/pull/6169

File: .rat-excludes [hunk 8]
```diff
@@ -0,0 +1,338 @@
...
+# .rat-excludes
+.rat-excludes
+# DISCLAIMER-WIP
+DISCLAIMER-WIP
+# drools-decisiontables/src/main/java/org/drools/decisiontable/parser/csv/CsvLineParser.java
+CsvLineParser.java
```
Comment (tkobayas) [inline .rat-excludes:8]: CsvLineParser.java is listed in `LICENSE`

File: .rat-excludes [hunk 9]
```diff
@@ -0,0 +1,338 @@
...
+# drools-decisiontables/src/test/resources/org/drools/decisiontable/unicode.drl.csv
+unicode.drl.csv
+# drools-decisiontables/src/test/resources/org/drools/decisiontable/integrationtests/csv/kmoduleCSVTest.dtable.drl.csv
+kmoduleCSVTest.dtable.drl.csv
+# drools-docs/supplemental-ui/css/search.css
+search.css
+# drools-docs/supplemental-ui/js/search-ui.js
+search-ui.js
+# drools-docs/supplemental-ui/js/vendor/lunr-2.3.9.min.js
+lunr-2.3.9.min.js
```
Comment (tkobayas) [inline .rat-excludes:34]: `drools-docs` is excluded from the source distribution, so no need to check.

File: .rat-excludes [hunk 10]
```diff
@@ -0,0 +1,338 @@
...
+# drools-drlonyaml-parent/drools-drlonyaml-todrl/src/main/resources/drl.ftl
+drl.ftl
+# drools-examples/src/main/resources/org/drools/examples/sudoku/Sample.sdo
+Sample.sdo
+# drools-examples/src/main/resources/org/drools/examples/troubleticketWithDSL/TroubleTicketWithDSL.dslr
+TroubleTicketWithDSL.dslr
+# drools-impact-analysis/drools-impact-analysis-graph/drools-impact-analysis-graph-graphviz/src/test/resources/org/drools/impact/analysis/graph/graphviz/simple.dot
+simple.dot
+# drools-model/drools-mvel-parser/src/main/javacc/mvel.jj
+mvel.jj
```
Comment (tkobayas) [inline .rat-excludes:50]: `mvel.jj` is listed in `LICENSE`

File: .rat-excludes [hunk 11]
```diff
@@ -0,0 +1,338 @@
...
+# kie-dmn/kie-dmn-backend/src/test/resources/DMN14.xsd
+DMN14.xsd
+# kie-dmn/kie-dmn-backend/src/test/resources/DMN15.xsd
+DMN15.xsd
+# kie-dmn/kie-dmn-backend/src/test/resources/DMNDI12.xsd
+DMNDI12.xsd
+# kie-dmn/kie-dmn-backend/src/test/resources/DMNDI13.xsd
+DMNDI13.xsd
+# kie-dmn/kie-dmn-backend/src/test/resources/DMNDI15.xsd
+DMNDI15.xsd
```
Comment (tkobayas) [inline .rat-excludes:116]: xsd files from OMG are listed in `LICENSE`

File: .rat-excludes (hunk 5 above)
Comment (jomarko) [reply .rat-excludes]: cool, thank you!

File: NOTICE (hunk 2 above)
Comment (jomarko) [reply NOTICE:23]: ok

File: .rat-excludes [hunk 12]
```diff
@@ -0,0 +1,338 @@
...
+# drools-drlonyaml-parent/drools-drlonyaml-cli-tests/src/test/resources/expected/test2.drl.txt
+test2.drl.txt
+# drools-drlonyaml-parent/drools-drlonyaml-cli-tests/src/test/resources/given/test1.drl.txt
+test1.drl.txt
+# drools-drlonyaml-parent/drools-drlonyaml-todrl/src/main/resources/drl.ftl
+drl.ftl
+# drools-examples/src/main/resources/org/drools/examples/sudoku/Sample.sdo
+Sample.sdo
+# drools-examples/src/main/resources/org/drools/examples/troubleticketWithDSL/TroubleTicketWithDSL.dslr
+TroubleTicketWithDSL.dslr
```
Comment (jomarko) [inline .rat-excludes]: It seems this file support comments, are we sure we can not add license? Does added license break the build?

File: .rat-excludes [hunk 13]
```diff
@@ -0,0 +1,338 @@
...
+# kie-pmml-trusty/kie-pmml-compiler/kie-pmml-compiler-commons/src/test/resources/KiePMMLTransformationDictionaryFactoryTest_01.txt
+KiePMMLTransformationDictionaryFactoryTest_01.txt
+# kie-pmml-trusty/kie-pmml-compiler/kie-pmml-compiler-commons/src/test/resources/KiePMMLTruePredicateFactoryTest_01.txt
+KiePMMLTruePredicateFactoryTest_01.txt
+# kie-pmml-trusty/kie-pmml-compiler/kie-pmml-compiler-commons/src/test/resources/TargetFieldFactoryTest_01.txt
+TargetFieldFactoryTest_01.txt
+# kie-pmml-trusty/kie-pmml-compiler/kie-pmml-compiler-commons/src/test/resources/TargetValueFactoryTest_01.txt
+TargetValueFactoryTest_01.txt
+# kie-pmml-trusty/kie-pmml-compiler/kie-pmml-compiler-core/PMMLCompiler.uml
+PMMLCompiler.uml
```
Comment (jomarko) [inline .rat-excludes]: I think it is just xml, would adding license as comment break the build?

**Review by jomarko — COMMENTED** [review]
@tkobayas thank you, I think last two comments/questions inline. I was focused for files, that ore not test files.

File: .rat-excludes (hunk 13 above)
Comment (tkobayas) [reply .rat-excludes]: Hi @gitgabrio , Could you confirm what license is applied to the `PMMLCompiler.uml` file? (e.g. what tool was used to export it. IntelliJ IDEA built-in tools?). Thanks!

File: .rat-excludes (hunk 12 above)
Comment (tkobayas) [reply .rat-excludes]: Thanks! Adding the headers and removing them from `.rat-excludes`

File: .rat-excludes (hunk 13 above)
Comment (tkobayas) [reply .rat-excludes]: At the moment, I assume that it's an usual output of IDE, I'm adding the apache license header.

**Comment by tkobayas** [comment]
Thank you @jomarko , I'll do the same fix to the main PR.

**Comment by gitgabrio** [comment]
Thanks @tkobayas 
FYI, the PMMLCompiler.uml file already has the license header, in [main](https://github.com/apache/incubator-kie-drools/blob/68561d8dbac8c4b723f19576e750d576dfd7a00c/kie-pmml-trusty/kie-pmml-compiler/kie-pmml-compiler-core/PMMLCompiler.uml).

**Review by jomarko — APPROVED** [review]
Thank you @tkobayas

**Pull request merged**